packages/callbacks/lib/tizen_interop_callbacks.dart

```diff
@@ -0,0 +1,348 @@
+       Pointer,  - user data for Dart callback
+       Pointer,  - pointer to name of multi proxy callback,
+                   allocated in Dart layer by toNativeUtf8() and freed by malloc.free,
+                   const char * in C++ layer
+       Int32     - proxy num
+       )
+    */
+    _registerWrappedCallback = _process
+        .lookup<
+            NativeFunction<
+                _RegistrationResult Function(Pointer, Bool, Pointer, Pointer,
+                    Int32)>>('TizenInteropCallbacksRegisterWrappedCallback')
+        .asFunction();
+
+    _unregisterWrappedCallback = _process
+        .lookup<NativeFunction<Pointer Function(Int32)>>(
+            'TizenInteropCallbacksUnregisterWrappedCallback')
+        .asFunction();
+
+    _platformProxyCallbackExists = _process
+        .lookup<NativeFunction<Bool Function(Pointer<Utf8>)>>(
+            'TizenInteropCallbacksProxyExists')
+        .asFunction();
+  }
+
+  void _callbackCallRequested(dynamic message) {
+    int callbackPtr = message;
+    final callback = Pointer<_CallbackWrapper>.fromAddress(callbackPtr);
+    _logDebug(_logTag,
+        'calling native runCallback(), cb=${callback.address.toRadixString(16)}');
+    _runCallbackInNativeLayer(callback);
+    _logDebug(_logTag, 'after calling native runCallback()');
+  }
+
+  int _getFirstFreeId(List<bool> list) {
+    return list.indexWhere((bool elem) => elem == false);
+  }
+
+  /// Registers a callbacks and returns wrapper/proxy data to be used
+  /// in Native API calls.
+  ///
+  /// To specify signature of wrapper callbacks, the name of Native API callback
+  /// defined in header file (such as `"device_changed_cb"`) must be provided
+  /// as [callbackName] parameter.
+  ///
+  /// [Pointer.fromFunction()] can be used to convert a Dart callback to ffi
+  /// pointer which is passed as [callback] parameter.
+  /// The requirements of fromFunction() applies - the function needs
+  /// to be static of free function. If the functions returns something, a fallback
+  /// return value needs to be specifed - a value which will be used in case
+  /// the Dart code invocation throws an exception.
+  ///
+  /// [NativeCbType] must match the signature of the Native API callback.
+  /// Proper type can be found in the relevant generated_binding.dart file,
+  /// look for the typedef for the callback.
+  /// The match is verified when [RegisteredCallback.interopCallback] is passed
+  /// to Native API.
+  ///
+  /// Most Tizen Native API callbacks offer passing arbitrary data
+  /// (in form of `void*`), known as `user_data`.
+  /// You can either specify [userData] - it will be passed to your Dart callback
+  /// as `user_data` parameter (if callback has such parameter), or [userObject] -
+  /// and then call [TizenInteropCallbacks.getUserObject(userData)] with the
+  /// obtained `user_data` value to retrieve the object back.
+  /// If neither is given, the `0`/`nullptr` will be passed.
```

Comment:
I was thinking that this solution belongs rather to interop/ffi world, and raw user_data was more fitting. But changing that will simplify the implementation a little. In the example apps I don't use the raw pointer, so I guess that shows it's not very useful anyway.

Reply:
There can be many design choices here. My initial suggestion was to implement the mapping in the user code (the example app) rather than in the plugin itself. The current implementation is also good. It's your choice.